### PR TITLE
Fix possible null value in AuthenticationSuccessHandler

### PR DESCRIPTION
**Pull request by bytehead**
See #1263.

### CODE REVIEW

**Comment by bytehead** [comment]
I don't know if it's really an issue - there was an old template in use in #1263. WDYT?

**Comment by leofeyer** [comment]
I'm not sure if it is wise to work around this – at least the error message makes the users aware that there is a problem. What happens if we silently ignore an empty target path?

**Comment by bytehead** [comment]
The user will possibly be redirected to the root page. Which is wrong at least for the backend.

**Comment by leofeyer** [comment]
Will they still realize that there is a problem in their template?

**Comment by bytehead** [comment]
Probably not. But the error from #1263 was anyway not too obvious imho.

**Comment by leofeyer** [comment]
@contao/developers WDYT?

**Comment by Toflar** [comment]
I think I would throw a 400 HTTP exception in this case. If it is a login request and it doesn't contain all the necessary information, that's a client request issue.

File: core-bundle/src/Security/Authentication/AuthenticationSuccessHandler.php [hunk 1]
```diff
@@ -168,4 +169,15 @@ private function triggerPostLoginHook(): void
             $system->importStatic($callback[0])->{$callback[1]}($this->user);
         }
     }
+
+    private function decodeTargetPath(Request $request): string
+    {
+        $targetPath = $request->request->get('_target_path');
+
+        if (null === $targetPath) {
+            throw new BadRequestHttpException('Missing form field `_target_path`');
```
Comment (Toflar) [inline core-bundle/src/Security/Authentication/AuthenticationSuccessHandler.php]:
```suggestion
            throw new BadRequestHttpException('Missing form field "_target_path". Did you maybe adjust any login template?');
```

Comment (bytehead) [reply core-bundle/src/Security/Authentication/AuthenticationSuccessHandler.php]: Fixed in f20a37cb0.

**Review by aschempp — DISMISSED** [review]

File: core-bundle/src/Security/Authentication/AuthenticationSuccessHandler.php [hunk 2]
```diff
@@ -168,4 +169,15 @@ private function triggerPostLoginHook(): void
             $system->importStatic($callback[0])->{$callback[1]}($this->user);
         }
     }
+
+    private function decodeTargetPath(Request $request): string
+    {
+        $targetPath = $request->request->get('_target_path');
+
+        if (null === $targetPath) {
+            throw new BadRequestHttpException('Missing form field "_target_path". Did you maybe adjust any login template?');
```
Comment (leofeyer) [inline core-bundle/src/Security/Authentication/AuthenticationSuccessHandler.php]:
```suggestion
            throw new BadRequestHttpException('Missing form field "_target_path". You probably need to adjust your custom login template.');
```

@bytehead What do you think?

Comment (bytehead) [reply core-bundle/src/Security/Authentication/AuthenticationSuccessHandler.php]: Fine for me 👍 Fixed in d034dac392021bddda6c530796d4e2877cdb8d71.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @bytehead.

**Pull request merged**